src/v2/cookbook/avoiding-memory-leaks.md

```diff
@@ -0,0 +1,138 @@
+  </div>
+</div>
+
+const app = new Vue({
+  el: "#app",
+  data: function() {
+    return {
+      showChoices: true
+    }
+  },
+  mounted: function() {
+    this.initializeChoices()
+  },
+  methods: {
+    initializeChoices() {
+      let list = []
+      // lets load up our select with a lot of choices so it will use a lot of memory
+      for (let i = 0; i < 1000; i++) {
+        list.push({
+          label: "Item " + i,
+          value: i
+        })
+      }
+      new Choices("#choices-single-default", {
+        searchEnabled: true,
+        removeItemButton: true,
+        choices: list
+      })
+    },
+    show() {
+      this.showChoices = true
+      this.$nextTick(() => {
+        this.initializeChoices()
+      })
+    },
+    hide() {
+      this.showChoices = false
+    }
+  }
+})
+
+```
+
+In the above example, we can use our `hide()` method to do some clean up and solve the memory leak prior to removing the select from the DOM. To accomplish this, we will keep a property in our Vue instance’s data object and we will use the [Choices API’s](https://github.com/jshjohnson/Choices) `destroy()` method to perform the clean up.
+
+Check the memory usage again with this [updated CodePen example](https://codepen.io/freeman-g/pen/mxWMor).
+
+```
+const app = new Vue({
+  el: "#app",
+  data: function() {
+    return {
+      showChoices: true,
+      choicesSelect: null
+    }
+  },
+  mounted: function() {
+    this.initializeChoices()
+  },
+  methods: {
+    initializeChoices() {
+      let list = []
+      for (let i = 0; i < 1000; i++) {
+        list.push({
+          label: "Item " + i,
+          value: i
+        })
+      }
+      // Set a reference to our choicesSelect in our Vue instance's data object
+      this.choicesSelect = new Choices("#choices-single-default", {
+        searchEnabled: true,
+        removeItemButton: true,
+        choices: list
+      })
+    },
+    show() {
+      this.showChoices = true
+      this.$nextTick(() => {
+        this.initializeChoices()
+      })
+    },
+    hide() {
+      // now we can use the reference to Choices to perform clean up here 
+      // prior to removing the elements from the DOM
+      this.choicesSelect.destroy()
+      this.showChoices = false
+    }
+  }
+})
+```
+
+## Additional Context
+
+In the above example, we used a `v-if` directive to illustrate the memory leak, but a more common real-world example happens when using **vue-router** to route to components in a Single Page Application.
+
+Just like the v-if directive, vue-router removes elements from the virtual DOM and replaces those with new elements when a user navigates around your application. The Vue `beforeDestroy()` [lifecycle hook](https://vuejs.org/v2/guide/instance.html#Lifecycle-Diagram) is a good place to solve the same sort of issue in a vue-router based application.
+
+We could move our clean up into the `beforeDestroy()` hook like this:
+
+```
+mounted: function() {
```

Comment:
I'm not sure we need `mounted` here. I guess it does provide a bit more context, but it seems a little weird to me with nothing in it. The three dots above `beforeDestroy` could probably be enough.